src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,57 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
 public class WorkWithFile {
+    private static final String SUPPLIED = "supply";
+    private static final String BOUGHT = "buy";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String dataFromFile = readFile(fromFileName);
+        String report = createReport(dataFromFile);
+        writeToFile(report, toFileName);
+    }
+
+    private String readFile(String fromFileName) {
+        StringBuilder dataFromFile = new StringBuilder();
+        String line;
+        try (BufferedReader reader = Files.newBufferedReader(Path.of(fromFileName))) {
+            while ((line = reader.readLine()) != null) {
+                dataFromFile.append(line).append(System.lineSeparator());
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read file " + fromFileName, e);
+        }
+        return dataFromFile.toString();
```

Comment:
Move return statement inside try statement
```suggestion
        try (BufferedReader reader = Files.newBufferedReader(Path.of(fromFileName))) {
            while ((line = reader.readLine()) != null) {
                dataFromFile.append(line).append(System.lineSeparator());
            }
            return dataFromFile.toString();
        } catch (IOException e) {
            throw new RuntimeException("Can't read file " + fromFileName, e);
        }
     
```